packages/stex-react-renderer/src/lib/ConceptHistoryTable.tsx

```diff
@@ -17,8 +19,34 @@ import {
   getConceptHistory,
 } from '@stex-react/api';
 import { useEffect, useState } from 'react';
+import { Chart } from 'react-google-charts';
 import { getMMTHtml } from './CompetencyTable';
 import { mmtHTMLToReact } from './mmtParser';
+import OpenInNewIcon from '@mui/icons-material/OpenInNew';
+
+const eventTypeMapping: { [key: string]: string } = {
+  'course-init': 'Previous Course Data',
+  'self-assessment': 'Self Assessment',
+  'self-assessment-5StepLikertSmileys': 'Self Assessment (Smileys)',
+  purge: 'Purge',
+  'i-know': 'I Know',
+  'problem-answer': 'Problem Answered',
+};
+
+function getProblemUrl(url: string): string {
+  const parts = url.split('/');
+
+  const startIndex = 3;
+  const endIndex = parts.indexOf('problems') + 1;
```

Comment:
handle the case when 'problems' is not found (`indexif('problems') === -1`)

Set `endIndex = startIndex + 2`